### PR TITLE
Simulator NavMesh visualization

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

Users may want to visualize the NavMesh loaded/modified in python. Also useful for tutorial/demo visualizations. The PR adds a `toggle_navmesh_visualization` function to the Simulator class to visualize a wireframe of the current `Simulator::pathfinder_` object.

Supporting changes:
 - Added new default material: `WHITE_MATERIAL_KEY = "white_ambient"` for wiremesh primitive visualizations.
 - `ResourceManager::primtive_meshes_` refactored to map and added `ResourceManager::removePrimitiveMesh(id)` to avoid stale asset build-up.
 - Refactored `Viewer` utility to better reflect these changes.

## How Has This Been Tested

 - Called in pytest (no verification).
 - Demo of feature in preliminary tutorial file `examples/tutorials/navmesh_tutorial.py` (This can be removed before merge until a better tutorial can be generated after colab/utils process is stable).

Example output:

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/sim/Simulator.cpp [hunk 1]
```diff
@@ -638,10 +638,37 @@ bool Simulator::recomputeNavMesh(nav::PathFinder& pathfinder,
...
+    return;
+  if (navMeshVisNode_ == nullptr && pathfinder_->isLoaded()) {
+    auto& sceneGraph = sceneManager_->getSceneGraph(activeSceneID_);
+    auto& rootNode = sceneGraph.getRootNode();
+    auto& drawables = sceneGraph.getDrawables();
+    navMeshVisNode_ = &rootNode.createChild();
+    navMeshVisPrimID_ = resourceManager_->loadNavMeshVisualization(
+        *pathfinder_, navMeshVisNode_, &drawables);
+    // navMeshVisNode_->translate({0, 0.1, 0}); //hover above the floor
+  } else if (navMeshVisNode_ != nullptr) {
```
Comment (erikwijmans) [inline src/esp/sim/Simulator.cpp:664]: This clean-up logic also needs to be in `close()`

File: src/esp/sim/Simulator.cpp [hunk 2]
```diff
@@ -638,10 +640,37 @@ bool Simulator::recomputeNavMesh(nav::PathFinder& pathfinder,
...
 
+void Simulator::toggleNavMeshVisualization() {
+  if (pathfinder_ == nullptr)
+    return;
+  if (navMeshVisNode_ == nullptr && pathfinder_->isLoaded()) {
+    auto& sceneGraph = sceneManager_->getSceneGraph(activeSceneID_);
+    auto& rootNode = sceneGraph.getRootNode();
+    auto& drawables = sceneGraph.getDrawables();
+    navMeshVisNode_ = &rootNode.createChild();
+    navMeshVisPrimID_ = resourceManager_->loadNavMeshVisualization(
```
Comment (eundersander) [inline src/esp/sim/Simulator.cpp:663]: I see that loadNavMeshVisualization can return ID_UNDEFINED. I guess removePrimitiveMesh will fail a CHECK on the next toggle. Does that sound like the behavior we want?

File: src/utils/viewer/viewer.cpp [hunk 3]
```diff
@@ -465,15 +473,18 @@ void Viewer::wiggleLastObject() {
 }
 
 void Viewer::toggleNavMeshVisualization() {
```
Comment (eundersander) [inline src/utils/viewer/viewer.cpp:475]: Would be nice if we didn't have to duplicate navMeshVisPrimID_, navMeshVidNode_, and toggleNavMeshVisualization in Viewer and Simulator.

File: examples/tutorials/navmesh_tutorial.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+from habitat_sim.utils import common as ut
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+data_path = os.path.join(dir_path, "../../data")
+output_path = os.path.join(dir_path, "navmesh_tutorial_output/")
+
+save_index = 0
+
+
+def make_video_cv2(observations, prefix="", open_vid=True, multi_obs=False):
```
Comment (eundersander) [inline examples/tutorials/navmesh_tutorial.py:21]: As you mentioned, it might be wise to omit this tutorial from this PR since Aaron is refactoring here.

**Comment by Skylion007** [comment]
Now that #677 is merged, maybe now is the time to bite the bullet and convert it to notebook form.

File: examples/tutorials/navmesh_tutorial.py (hunk 4 above)
Comment (Skylion007) [inline examples/tutorials/navmesh_tutorial.py:21]: We should extract this and some other code out and put it in viz_utils. Also a good time to start turning them into notebooks.

File: src/esp/sim/Simulator.cpp (hunk 2 above)
Comment (aclegg3) [reply src/esp/sim/Simulator.cpp:663]: Good point, thanks. 👍

File: src/utils/viewer/viewer.cpp (hunk 3 above)
Comment (aclegg3) [reply src/utils/viewer/viewer.cpp:475]: Agreed. We need to refactor Viewer to use the Simulator class and this kind of duplication won't be a problem anymore. 🤷‍♂️

File: examples/tutorials/navmesh_tutorial.py (hunk 4 above)
Comment (aclegg3) [reply examples/tutorials/navmesh_tutorial.py]: Great point, let's cache this for a conversion PR.

File: src/utils/viewer/viewer.cpp (hunk 3 above)
Comment (dhruvbatra) [reply src/utils/viewer/viewer.cpp:475]: I know I'm late to this conversation, but completely agree.

**Pull request merged**